tests/conftest.py

```diff
@@ -159,3 +173,210 @@ def zarr_format(request: pytest.FixtureRequest) -> ZarrFormat:
     suppress_health_check=[HealthCheck.filter_too_much, HealthCheck.too_slow],
     verbosity=Verbosity.verbose,
 )
+
+# TODO: uncomment these overrides when we can get mypy to accept them
+"""
+@overload
+def create_array_metadata(
+    *,
+    shape: ShapeLike,
+    dtype: npt.DTypeLike,
+    chunks: ChunkCoords | Literal["auto"],
+    shards: None,
+    filters: FiltersLike,
+    compressors: CompressorsLike,
+    serializer: SerializerLike,
+    fill_value: Any | None,
+    order: MemoryOrder | None,
+    zarr_format: Literal[2],
+    attributes: dict[str, JSON] | None,
+    chunk_key_encoding: ChunkKeyEncoding | ChunkKeyEncodingLike | None,
+    dimension_names: None,
+) -> ArrayV2Metadata: ...
+
+
+@overload
+def create_array_metadata(
+    *,
+    shape: ShapeLike,
+    dtype: npt.DTypeLike,
+    chunks: ChunkCoords | Literal["auto"],
+    shards: ShardsLike | None,
+    filters: FiltersLike,
+    compressors: CompressorsLike,
+    serializer: SerializerLike,
+    fill_value: Any | None,
+    order: None,
+    zarr_format: Literal[3],
+    attributes: dict[str, JSON] | None,
+    chunk_key_encoding: ChunkKeyEncoding | ChunkKeyEncodingLike | None,
+    dimension_names: Iterable[str] | None,
+) -> ArrayV3Metadata: ...
+"""
+
+
+def create_array_metadata(
```

Comment:
seems like this would duplicate logic form elsewhere? Is there nowhere else this function could be used?

Reply:
this _should_ be duplicated, but sadly these functions don't exist in the codebase yet! I add some functions like this in another PR: https://github.com/zarr-developers/zarr-python/pull/2761